internal/mikrotik/types.go

```diff
@@ -1,12 +1,20 @@
 package mikrotik
 
+import (
+	"math"
+	"time"
+
+	"sigs.k8s.io/external-dns/endpoint"
+)
+
 // config details for authentication with the MikroTik RouterOS API
 type Config struct {
-	Host          string `env:"MIKROTIK_HOST,notEmpty"`
-	Port          string `env:"MIKROTIK_PORT,notEmpty" envDefault:"443"`
-	Username      string `env:"MIKROTIK_USERNAME,notEmpty"`
-	Password      string `env:"MIKROTIK_PASSWORD,notEmpty"`
-	SkipTLSVerify bool   `env:"MIKROTIK_SKIP_TLS_VERIFY" envDefault:"false"`
+	Host          string       `env:"MIKROTIK_HOST,notEmpty"`
+	Port          string       `env:"MIKROTIK_PORT,notEmpty" envDefault:"443"`
+	Username      string       `env:"MIKROTIK_USERNAME,notEmpty"`
+	Password      string       `env:"MIKROTIK_PASSWORD,notEmpty"`
+	SkipTLSVerify bool         `env:"MIKROTIK_SKIP_TLS_VERIFY" envDefault:"false"`
+	DefaultTTL    endpoint.TTL `env:"MIKROTIK_DEFAULT_TTL" envDefault:"3600"`
```

Comment:
I agree that the default should just let RouterOS determine it, but I do still want this env var to exist (in addition to the per-endpoint config which works too).

Reply:
Hm... since the TTL would be configurable per-endpoint, a workaround for this global env var would be to use something similar to a kyverno policy to mutate the objects to include that annotation.

It is a bit hacky, so I am considering an env var as a simpler solution, but if you already have Kyverno deployed in your environment, that would be an option.

Reply:
I don't see why that would be necessary, it's a relatively simple change. I just need to remove the `envDefault` from the config and only set `DNSRecord.TTL` if it's not an empty string.

Reply:
I've pushed an attempt at this in https://github.com/mirceanton/external-dns-provider-mikrotik/pull/59/commits/5d0c3dbd8584ff99e055bad6ad4ac01a5dbe9b19 but I haven't tested it at all. Without this commit external-dns works fine, but it's possible it won't be comparing records correctly anymore, since it always gets a TTL from the router but only sometimes sets it itself.

Reply:
I think an approach similar to https://github.com/mirceanton/external-dns-provider-mikrotik/commit/82826e8b776b5988779602252ba97db63724ea06 is also good here? I mean having the default set there